### ISSUE TITLE
UselessIfConditionWithReturn false positive in simple case

### ISSUE THREAD

**Issue by kkmuffme** [slevomat/coding-standard#1437]
>slevomat/coding-standard                       8.5.2

Config:

```
<?xml version="1.0"?>
<ruleset name="devste">
	<rule ref="SlevomatCodingStandard.ControlStructures.UselessIfConditionWithReturn"/>
</ruleset>
```

```php
<?php
namespace My_Classes;

/**
 * my class
 */
class Foo {
	/**
	 * do something
	 *
	 * @return bool
	 */
	public function has_xyz() {
		$x = call_this( 'x' );
		if ( empty( $x ) || !is_string( $x ) ) {
			return false;
		}

		$y = call_this( 'y' );
		if ( empty( $y['abc'] ) ) {
			return false;
		}

		return true;
	}
}
```

Reports `Useless condition.` error for `if ( empty( $y['abc'] ) ) {` - but this condition is not useless and cannot be removed.

I'm not 100% sure, but I think this bug was added recently, maybe even in 8.5.2 bc I can't remember having this in 8.5

**Comment by kukulich** [comment]
@kkmuffme I think the condition is really useless. Try the fixer and see the fixed code.

**Comment by roslov** [comment]
@kkmuffme, `return !empty($y['abc']);` is much better. So, the condition is definitely useless.
